README-design.md

```diff
@@ -0,0 +1,45 @@
+This document describes some of the current high level design
+of the project.  It assumes some familiarity with the primary `README.md`.
+
+Builds
+----
+
+Conceptually, each run of `coreos-assembler build` may generate a new "build".
+
+By default, a single build will generate both an OSTree commit and a `-qemu.qcow2`.
+The OSTree commit data is generated via rpm-ostree, using `src/config/manifest.yaml`.
+The QEMU image uses the OSTree commit and the `src/config/image.ks` data, which
+defines the partition layout.
+
+The new OSTree commit is written into `repo`; if you have configured a ref in
+the manifest, it will be updated.
+
+The core coreos-assembler concept of a "build" though is represented primarily by a
+new subdirectory in `builds/$version`, and inside that directory there's a
+`meta.json` that contains a lot of relevant metadata, including the OSTree commit.
+
+By default, builds are pruned, although one can use `build --no-prune` to
+prevent this.
+
+Change detection
+---
+
+All of the interesting content of a build goes into the ostree commit.  Images
```

Comment:
I think we have different ideas of "interesting". Better to be technically correct and phrase this as something like "All of the contents of filesystems generated by a build goes into...".

I also disagree that "images are just wrappers". Grub, partition layouts and the filesystems on disk matter and shouldn't be discounted.